python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py

```diff
@@ -1480,34 +1477,46 @@ def get_subset_selection_for_context(
-        for check_key in (all_check_keys - context.selected_asset_check_keys)
-    )
-
-    # note that this will always be true if checks are disabled
-    if selected_checks_target_selected_assets and all_checks_included_for_selected_assets:
-        selected_dbt_resources = selected_dbt_non_test_resources
-        indirect_selection = None
+    excluded_checks = all_check_keys - context.selected_asset_check_keys
+    excluded_checks_on_selected_assets = [
+        check_key
+        for check_key in excluded_checks
+        if check_key.asset_key in context.selected_asset_keys
+    ]
+
+    # note that this will always be false if checks are disabled (which means the assets_def has no
+    # check specs)
+    if checks_on_non_selected_assets or excluded_checks_on_selected_assets:
+        # select all assets and tests explicitly, and turn off indirect selection. This risks
+        # hitting the CLI argument length limit, but in the common scenarios that can be launched from the UI
+        # (all checks disabled, only one check and no assets) it's not a concern.
+        # Since we're setting DBT_INDIRECT_SELECTION=empty, we won't run any singular tests.
+        check.invariant(dagster_dbt_translator.settings.enable_asset_checks)
+        selected_dbt_resources = selected_dbt_non_test_resources + selected_dbt_tests
+        indirect_selection = DBT_EMPTY_INDIRECT_SELECTION
         logger.info(
             "A dbt subsetted execution is being performed. Overriding default dbt selection"
-            f" arguments `{default_dbt_selection}` with arguments: `{selected_dbt_resources}`. "
+            f" arguments `{default_dbt_selection}` with arguments: `{selected_dbt_resources}`."
+        )
+        logger.info(
+            "A subsetted execution for asset checks is being performed. Overriding default "
+            f"`DBT_INDIRECT_SELECTION` {env.get('DBT_INDIRECT_SELECTION', 'eager')} with "
+            f"`{indirect_selection}` due to additional checks "
+            f"{', '.join([c.to_user_string() for c in checks_on_non_selected_assets])} "
+            f"and excluded checks {', '.join([c.to_user_string() for c in excluded_checks_on_selected_assets])}."
         )
-    # otherwise we select all assets and tests explicitly, and turn off indirect selection. This risks
-    # hitting the CLI argument length limit, but in the common scenarios that can be launched from the UI
-    # (all checks disabled, only one check and no assets) it's not a concern.
-    # Since we're setting DBT_INDIRECT_SELECTION=empty, we won't run any singular tests.
+
     else:
-        check.invariant(dagster_dbt_translator.settings.enable_asset_checks)
-        selected_dbt_resources = selected_dbt_non_test_resources + selected_dbt_tests
-        indirect_selection = DBT_EMPTY_INDIRECT_SELECTION
+        selected_dbt_resources = selected_dbt_non_test_resources
+        indirect_selection = None
         logger.info(
```

Comment:
This logging statement is repeated. We should find a way to move this out of the conditional branch and into the main body of logic to simplify things.